index.js

```diff
@@ -299,9 +299,10 @@ RequireFilter.prototype.processString = function(code) {
   return replaceRequireAndDefine(code, this.amdPackages, this.amdModules);
 };
 
-function write(arr, str, offset) {
-  for (var i = 0, l = str.length; i < l; i++) {
-    arr[offset + i] = str[i];
+function write(arr, str, offset, len) {
+  arr[offset] = str;
+  for (let i = 1; i < len; i++) {
+    arr[offset + i] = undefined;
```

Comment:
I think the loop is wrong.
Is your intention to fill up the remaining of the space with `undefined`?
What you seem to try to do is:
- copy the `str` into the array at the `offset`
- then fill the *rest* with undefined
If this is true then the for loop should start at `str.lengh`
Add a comment to explain what you are trying to do.

Reply:
The loop is correct.

The idea is that before this function is called, the original unmodified code is stored between `offset` (range[0]) and `offset + len` (range[1]).

We need to fit the modified code into the same range. If the buffer size is changed, the remaining node.range from the ast are out of sync with the buffer... and it would be tricky to keep track of how things shifted.

So my little "hack" is the putting the entire string into the offset position, then fill the remaining characters in the original unmodified code range with undefined (node.range[0] + 1, node.range[1]). Since we are really just using `buffer.join('')` at the end. This works.

Here is an example:
```
const buffer = ['H', 'e', 'l', 'l', 'o', ' ', 'W', 'o', 'r', 'l', 'd'].
```
Let's say we want to replace "Hello" with "Goodnight"

In this case, `node.range = [0, 5]`, `str = "Goodnight"`.

So offset = node.range[0] = 0, len = 5

Using my method, the buffer becomes:
```
buffer = ['Goodnight', undefined, undefined, undefined, undefined, ' ', 'W', 'o', 'r', 'l', 'd']
```

Using you suggestion it would become:
```
buffer = ['Goodnight', undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, 'l', 'd'];
```
which is not what we want.
```


I thought about actually just passing the range object to this function.
So it would look like:
```
function write(arr, str, range) {
  const offset = range[0];
  arr[offset] = str;
  for (let i = 1; i < range[1]; i++) {
    arr[offset + i] = undefined;
  }
}
```

Let me know if you prefer my alternative. I will also leave a comment

Reply:
Hopefully my explanation above helped you understand the logic.

Also, updated the write signature and added a comment explaining the write function "hack"

Reply:
OK. Now I understand what you are doing. 
Pretty cool/smart.

Is this working in your app?

Reply:
Yes. Everything here has been tested against my application!